solr/core/src/java/org/apache/solr/cloud/RecoveryStrategy.java

```diff
@@ -911,18 +911,39 @@ private final void sendPrepRecoveryCmd(String leaderBaseUrl, String leaderCoreNa
     int readTimeout =
         conflictWaitMs
             + Integer.parseInt(System.getProperty("prepRecoveryReadTimeoutExtraWait", "8000"));
-    try (HttpSolrClient client =
+    var recoveryExec =
+        ExecutorUtil.newMDCAwareFixedThreadPool(
+            1, new SolrNamedThreadFactory("sendPrepRecoveryCmd"));
```

Comment:
shutdown waits for completion; this may not be obvious but it's pertinent to the logic.  A comment should shine a light on this

Reply:
Come to think of it, maybe the executor should be the field so is re-used?